### ISSUE TITLE
Assets disappeared please help.

### ISSUE THREAD

**Issue by INeedHelp225** [SmartlyDressedGames/Unturned-3.x-Community#2826]
I have been developing a map for 5 months at this point. 2 weeks ago on august 31st I played for the last time, taking a two week break because of work and school, and lack of interest. I now have time and have come back to find my map ruined. Nearly all Hawaii and Greece asset has disappeared from the map:
![image](https://user-images.githubusercontent.com/59327136/133871792-89b16353-acc1-431e-9e8f-6d8a83b02ec6.png)
strangely the bulldozer hasnt gone...
![image](https://user-images.githubusercontent.com/59327136/133871813-dd6ae1de-3287-4690-8661-152742102926.png)
before
![image](https://user-images.githubusercontent.com/59327136/133871823-bf28344a-0dc8-4473-9b4d-f17cd33069b3.png)
after

Another strange thing is that in an ancient backup, all the Greece and Hawaii assets remain there, nothing gone. It makes no sense.
I haven't yet saved this hellhole ruined version of my map, is there any way to save it? I've spent so so long on it at this point...

**Comment by SDGNelson** [comment]
If you look in the Menu > Workshop > Manage Subscriptions menu do Hawaii and Greece show up still?

After loading into the editor again could you attach the Client.log file from the Unturned/Logs folder please?

**Comment by INeedHelp225** [comment]
![image](https://user-images.githubusercontent.com/59327136/133914813-3d04e2aa-2a15-411a-a180-338726026e4f.png)
Indeed it is, they are present. I have also made sure install and re-install them several times, but the problem remains.

Client.log file for the broken map
[Client.log](https://github.com/SmartlyDressedGames/Unturned-3.x-Community/files/7190956/Client.log)

Apologies for all the "X is taken by Y" spam, i have legacy Ireland and modern Ireland assets both installed.

**Comment by SDGNelson** [comment]
If you load Hawaii and Greece are the assets missing?

According to the log the only missing assets are Reactor_0, Metro_Bale_1, and Torture_3.

Out of curiosity, how did you take those before/after screenshots so precisely? Did you load up an older version of the map and they still loaded fine? That would suggest to me the objects really have been deleted from the map rather than missing the assets.

Perhaps a version of your map was loaded from the workshop rather than from local files?

Perhaps the map was saved with `-NoPreserveMissingObjects` at some point?

**Comment by INeedHelp225** [comment]
The Hawaii and Greece assets are gone from the map, excluding a few strange things like the aforementioned bulldozers, cave fungus mushrooms, waterpark assets, and the observatory asset. However, on a test map, an abandoned project, and the old backup, the Hawaii and Greece assets remain unchanged. 

The preciseness of the screenshots is just a coincidence, I took the original picture on the 29th of August to show to a friend, and the secondary photo is just me coming back to the map in a similar spot to show the missing objects. The current version of the map had the bridge and a lot more, until it all disappeared.
I haven't played since the 31st on the map, I took a break to play Sim City 4 and work until September 14th. That's when I noticed the assets gone, the map is still unsaved from that point:
![image](https://user-images.githubusercontent.com/59327136/133936684-c11ed6b4-d596-47e4-ac94-403101e000f6.png)

I haven't uploaded the map to the workshop yet, and I am unknowing of the "-NoPreserveMissingObjects" tag.

This is a really confusing mystery, like you said, it appears the objects have been deleted, but I did not delete anything. I don't know why only a few Hawaii and Greece assets would disappear in the map, and not Easter Island or France or any other curated workshop add-on.  This whole thing is very frustrating.

**Comment by SDGNelson** [comment]
Since the objects are visible in other maps I am sorry to say it does sound like they have been deleted from your map. If you would like to email me the map I could try loading it up to double-check whether there is any record of those objects. Otherwise do you have any backups to restore from?

**Comment by INeedHelp225** [comment]
I've gone ahead and sent the map, along with the old, asset-filled backup, to your nelson@smartlydressedgames.com address. The latest backup I've got is from 9 days prior, and a lot of work had occurred in-between those 9 days. I suppose I've learned my lesson on backing up, lol.

**Comment by SDGNelson** [comment]
Thanks for sending over the map! This evening I loaded it up and am sorry to say the objects do seem to be completely removed. I added some extra checks to the editor for objects with NaN/infinite position/scale, position outside bounds, nearly zero scale, etc to see if those had somehow been corrupted but no luck.

Sorry not to have a good answer. For what it's worth, I would recommend using version control software like Git in the future. Much easier than manually keeping backups and gives you a detailed history to check back on.
